src/bin/casr-cluster.rs

```diff
@@ -99,18 +124,13 @@ fn stacktrace(path: &Path) -> Result<Stacktrace> {
                     trace.push(entry.to_string());
                 }
             });
-            let mut rawtrace = if let Some(asan_array) = u.get("AsanReport") {
-                if let Some(asan_array) = asan_array.as_array() {
-                    if asan_array.iter().next().is_some() {
-                        casr::asan::stacktrace_from_asan(&trace)
-                            .with_context(|| format!("File: {}", path.display()))?
-                    } else {
-                        Stacktrace::from_gdb(&trace.join("\n"))
-                            .with_context(|| format!("File: {}", path.display()))?
-                    }
-                } else {
-                    bail!("Error while parsing AsanReport. File: {}", path.display());
-                }
+
+            let mut rawtrace = if is_not_empty_as_array(&u, "AsanReport", path)? {
+                casr::asan::stacktrace_from_asan(&trace)
+                    .with_context(|| format!("File: {}", path.display()))?
+            } else if is_not_empty_as_array(&u, "PythonReport", path)? {
```

Comment:
It's better just write code 2 time here than use this function with strange semantics.

Reply:
Done